src/Rematch.jl

```diff
@@ -128,42 +128,76 @@ function handle_destruct(value::Symbol, pattern, bound::Set{Symbol}, asserts::Ve
             end
         end
     elseif @capture(pattern, T_(subpatterns__))
-        # struct      
-        len = length(subpatterns)
-        named_fields = [pat.args[1] for pat in subpatterns if (pat isa Expr) && pat.head == :kw]
-        named_count = length(named_fields)
-        @assert named_count == length(unique(named_fields)) "Pattern $pattern has duplicate named arguments ($(named_fields))."
-        @assert named_count == 0 || named_count == len "Pattern $pattern mixes named and positional arguments."
-        if named_count == 0
-            # Pattern uses positional arguments to refer to fields e.g. Foo(0, true)
-            expected_fieldcount = gensym("$(T)_expected_fieldcount")
-            actual_fieldcount = gensym("$(T)_actual_fieldcount")
-            push!(asserts, quote
-                if typeof($(esc(T))) <: Function
-                    throw(LoadError("Attempted to match on a function", @__LINE__, AssertionError("Incorrect match usage")))
+        # struct or extractor call
+        if !isempty(string(T)) && islowercase(first(string(T)))
+            # Extractor call.
+            result = gensym("unapply")
+            len = length(subpatterns)
+            # Extractor call.
+            # The function should take one argument and return either `nothing`, if the argument does not match,
+            # or a tuple if it does match.
```

Comment:
Please make sure to also break comments at 92chars to stay consistent with the style guide.

Reply:
Line lengths fixed throughout the file.